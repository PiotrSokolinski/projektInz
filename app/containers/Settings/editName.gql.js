import gql from 'graphql-tag'

export default gql`
  mutation editName($firstName: String!, $lastName: String!) {
    editName(firstName: $firstName, lastName: $lastName) {
      id
      firstName
      lastName
    }
  }
`
