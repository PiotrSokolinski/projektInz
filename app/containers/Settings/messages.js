/*
 * Settings Messages
 *
 * This contains all the text for the Settings container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.Settings'

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Account Settings',
  },
  firstNameLabel: {
    id: `${scope}.firstNameLabel`,
    defaultMessage: 'First Name',
  },
  lastNameLabel: {
    id: `${scope}.lastNameLabel`,
    defaultMessage: 'Last Name',
  },
  nickLabel: {
    id: `${scope}.nickLabel`,
    defaultMessage: 'Nick',
  },
  editPhoto: {
    id: `${scope}.editPhoto`,
    defaultMessage: 'Edit Profile Photo',
  },
  firstNameEmptyError: {
    id: `${scope}.firstNameEmptyError`,
    defaultMessage: 'First name can not be empty',
  },
  lastNameEmptyError: {
    id: `${scope}.lastNameEmptyError`,
    defaultMessage: 'Last name can not be empty',
  },
  editDetails: {
    id: `${scope}.editDetails`,
    defaultMessage: 'Edit details',
  },
  submitChanges: {
    id: `${scope}.submitChanges`,
    defaultMessage: 'Submit changes',
  },
  changeMailTitle: {
    id: `${scope}.changeMailTitle`,
    defaultMessage: 'Chagne your e-mail address',
  },
  chagnePasswordTitle: {
    id: `${scope}.chagnePasswordTitle`,
    defaultMessage: 'Chagne your password',
  },
  changeMail: {
    id: `${scope}.changeMail`,
    defaultMessage: 'Change e-mail',
  },
  changePassword: {
    id: `${scope}.changePassword`,
    defaultMessage: 'Change password',
  },
  changePasswordTitle: {
    id: `${scope}.changePasswordTitle`,
    defaultMessage: 'Set a new password',
  },
  setMailTitle: {
    id: `${scope}.changeMailTitle`,
    defaultMessage: 'Set a new e-mail address',
  },
  userCredentials: {
    id: `${scope}.userCredentials`,
    defaultMessage: `User's Credentials`,
  },
})
