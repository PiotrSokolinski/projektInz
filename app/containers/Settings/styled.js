import styled from 'styled-components'

import Input from 'components/Input'
import Button from 'components/Button'
import { Colors } from 'themes'

export const EditPhotoButton = styled(Button)`
  margin-left: 25px;
`

export const EditButton = styled(Button)`
  margin-top: 15px;
  width: 120px;
  height: 30px;
`

export const ChangeButton = styled(Button)`
  width: 200px;
  height: 50px;
  margin-top: 35px;
`

export const Container = styled.div`
  display: flex;
  height: calc(100vh - 75px);
`

export const ContentWrapper = styled.div`
  margin: 30px;
  width: 100%;
  background: ${Colors.concrete};
  border-radius: 5px;
`

export const Header = styled.div`
  font-size: 25px;
  font-weight: bold;
  margin: 25px 25px 0;
  text-align: center;
`

export const PhotoWrapper = styled.div`
  display: flex;
  align-items: center;
  border-bottom: 1px solid ${Colors.edward};
  padding: 0 30px 20px;
`

export const InformationWrapper = styled.div`
  display: flex;
  margin: 30px;
`

export const DetailsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 25px;
  width: 300px;
`

export const AccoutInput = styled(Input)`
  margin-top: 25px;
`

export const ChangeHeader = styled.div`
  font-size: 19px;
  margin-top: 25px;
  font-weight: bold;
  text-align: center;
`

export const CredentialsContainer = styled.div`
  margin-left: 100px;
  display: flex;
  flex-direction: column;
  align-items: center;
`
