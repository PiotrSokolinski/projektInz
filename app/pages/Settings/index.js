import React from 'react'
import SettingsContainer from 'containers/Settings'
const Settings = props => <SettingsContainer {...props} />

export default Settings
