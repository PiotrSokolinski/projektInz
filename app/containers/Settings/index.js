/**
 *
 * Settings
 *
 */

import React from 'react'
// import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'

import messages from './messages'

const Settings = () => (
  <div>
    <FormattedMessage {...messages.header} />
  </div>
)

Settings.propTypes = {}

export default Settings
