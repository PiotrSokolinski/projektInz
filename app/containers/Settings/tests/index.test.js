/**
 *
 * Tests for Settings
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import Settings from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<Settings />', () => {
  it.skip('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Settings />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Settings />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
