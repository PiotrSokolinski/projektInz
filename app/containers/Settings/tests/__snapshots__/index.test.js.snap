// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> Should render and match the snapshot 1`] = `
<div>
  <span>
    This is the Settings container!
  </span>
</div>
`;
